test/test_queue.py

```diff
@@ -694,3 +694,71 @@ def test_throttled_additional_put(self):
         self.lua('put', 2, 'worker', 'queue', 'jid2', 'klass', {}, 0)
         self.assertEqual(self.lua('throttle.locks', 3, 'ql:q:queue'), ['jid1'])
         self.assertEqual(self.lua('jobs', 4, 'throttled', 'queue'), ['jid2'])
+
+    def test_pop_no_retry(self):
```

Comment:
Might be missing it but don't see the tests assert that the job was properly moved. It should be tested elsewhere but be nice to see it.